src/components/form/select.js

```diff
@@ -1,80 +1,364 @@
   SelectVariant
 } from '@patternfly/react-core';
 import _cloneDeep from 'lodash/cloneDeep';
-import _isEqual from 'lodash/isEqual';
 import _findIndex from 'lodash/findIndex';
 import _isPlainObject from 'lodash/isPlainObject';
-import { helpers } from '../../common/helpers';
+import _memoize from 'lodash/memoize';
+import { createMockEvent } from './formHelpers';
+import { helpers } from '../../common';
+
+/**
+ * Dropdown split button variants
+ *
+ * @type {{action: string, checkbox: string}}
+ */
+const SplitButtonVariant = {
+  action: 'action',
+  checkbox: 'checkbox'
+};
+
+/**
+ * Dropdown toggle button variants
+ *
+ * @type {{secondary: string, default: string, plain: string, text: string, primary: string}}
+ */
+const ButtonVariant = PfButtonVariant;
+
+/**
+ * Pass button variant as a select component option.
+ *
+ * @type {{secondary: string, default: string, plain: string, text: string, primary: string}}
+ */
+const SelectButtonVariant = ButtonVariant;
 
 /**
  * Pass direction as select component variant option.
  *
- * @type {DropdownDirection}
+ * @type {{down: string, up: string}}
  */
 const SelectDirection = DropdownDirection;
 
 /**
  * Pass position as select component variant option.
  *
- * @type {DropdownPosition}
+ * @type {{left: string, right: string}}
  */
 const SelectPosition = DropdownPosition;
 
 /**
- * FixMe: PF isFlipEnabled busts left/right positioning, we now disable it by default
- * We caught this adjustment through our component code test snapshots. PF positioning for left/right has issues in general on
- * Select (Dropdown used to use the same styling, and functioned correctly, we'd need to reevaluate it now). We patched Select
- * with a minor styling adjustment for position from Dropdown, this worked until isFlipEnabled became defaulted to true.
- * Evaluate on next PF patch.
+ * FixMe: attributes on PF select and dropdown components do not allow data- attributes being passed
+ * Bypass pf with refs and apply the attributes.
  */
 /**
- * A wrapper for Patternfly Select. Provides restructured event data for onSelect callback.
+ * Format options into a consumable array of objects.
+ * Note: It is understood that for line 83'ish around "updatedOptions" we dump all values regardless
+ * of whether they are plain objects, or not, into updatedOptions. This has been done for speed only,
+ * one less check to perform.
  *
- * @augments React.Component
- * @fires onSelect
- * @fires onToggle
+ * @param {object} params
+ * @param {*|React.ReactNode} params.selectField
+ * @param {object|Array} params.options
+ * @param {string|number|Array} params.selectedOptions
+ * @param {string} params.variant
+ * @param {object} params.props
+ * @returns {{options: *[]|*, selected: *[]}}
  */
-class Select extends React.Component {
-  state = { isExpanded: false, options: null, selected: null };
+const formatOptions = ({ selectField = { current: null }, options, selectedOptions, variant, ...props } = {}) => {
```

Comment:
optional: review using a memoize on this format helper... it is understood that updating "selectedOptions" might make it useless

Reply:
looks like we already did a review on memoize, the notes indicate the gain was minimal, a human wouldn't notice the difference... but we'll check again

Reply:
the state update combined with the effect diff help prevent unnessary use... adding in the memoize, again, provides no noticeable gain other than a possible head scratch